File: hays_order_wa/services/pizza/pizzaService.js
import axios from "../axios";

const createPizza = (pizzaDto) => {
  return axios.post("/pizza", pizzaDto);
};

const getXPizzasByName = (name, limit) => {
  return axios.get(`/pizza/name/${name}/${limit}`);
};

const getAllPizzas = () => {
  return axios.get(`/pizza/all`);
};

const getPizzaById = (id) => {
  return axios.get(`/pizza/${id}`);
};

export { createPizza, getXPizzasByName, getPizzaById , getAllPizzas};

File: hays_order_ms/src/main/java/com/hays/hays_order_ms/application/port/in/PizzaService.java
package com.hays.hays_order_ms.application.port.in;

import com.hays.hays_order_ms.domain.model.Pizza;

import java.util.List;

public interface PizzaService {
    Pizza createPizza(Pizza pizza);
    List<Pizza> getXPizzasByName(String name, Integer number);
    List<Pizza> getAllPizzas();
    Pizza getPizzaById(Long id);
}
